Sources/Sentry/SentryPerformanceTracker.m

```diff
@@ -45,17 +45,28 @@ - (SentrySpanId *)startSpanWithName:(NSString *)name operation:(NSString *)opera
         activeSpan = [self.activeSpanStack lastObject];
     }
 
-    id<SentrySpan> newSpan;
+    __block id<SentrySpan> newSpan;
     if (activeSpan != nil) {
         newSpan = [activeSpan startChildWithOperation:operation description:name];
     } else {
         SentryTransactionContext *context =
             [[SentryTransactionContext alloc] initWithName:name operation:operation];
-        newSpan =
-            [SentrySDK.currentHub startTransactionWithContext:context
-                                                  bindToScope:SentrySDK.currentHub.scope.span == nil
-                                              waitForChildren:YES
-                                        customSamplingContext:@{}];
+
+        [SentrySDK.currentHub.scope useSpan:^(id<SentrySpan> span) {
+            BOOL bindToScope = true;
+            if (span != nil) {
+                if ([span.context.operation isEqualToString:@"ui.action"]) {
```

Comment:
Im not sure about comparing the operation here.
Maybe we could have some kind of span level to determined whether we can override the current one in the scope or not.
But possible a future refactoring.